File: Bridge/Resources/.generated/system/Text/encoding.js
    Bridge.define("System.Text.Encoding", {
        inherits: [System.ICloneable],
        statics: {
            props: {
                UTF8: {
                    get: function () {
                        throw new System.NotImplementedException();
                    }
                },
                UTF8UnmarkedUnsafe: {
                    get: function () {
                        throw new System.NotImplementedException();
                    }
                }
            }
        },
        alias: ["clone", "System$ICloneable$clone"],
        ctors: {
            ctor: function () {
                this.$initialize();
            }
        },
        methods: {
            getBytes$4: function (s, charIndex, charCount, bytes, byteIndex) {
                throw new System.NotImplementedException();
            },
            getBytes$2: function (s) {
                throw new System.NotImplementedException();
            },
            getBytes$1: function (chars, index, count) {
                throw new System.NotImplementedException();
            },
            getBytes: function (chars) {
                throw new System.NotImplementedException();
            },
            getString$1: function (bytes, index, count) {
                throw new System.NotImplementedException();
            },
            getString: function (bytes) {
                throw new System.NotImplementedException();
            },
            getDecoder: function () {
                throw new System.NotImplementedException();
            },
            clone: function () {
                throw new System.NotImplementedException();
            }
        }
    });

File: Bridge/System/Text/Encoding.cs
using Bridge;

namespace System.Text
{
    public abstract class Encoding : ICloneable
    {
        protected Encoding()
        {
        }

        public abstract int GetBytes(char[] chars, int charIndex, int charCount, byte[] bytes, int byteIndex);

        public virtual int GetBytes(String s, int charIndex, int charCount, byte[] bytes, int byteIndex)
        {
            throw new NotImplementedException();
        }
        
        public virtual byte[] GetBytes(String s)
        {
            throw new NotImplementedException();
        }

        public virtual byte[] GetBytes(char[] chars, int index, int count)
        {
            throw new NotImplementedException();
        }

        public virtual byte[] GetBytes(char[] chars)
        {
            throw new NotImplementedException();
        }

        public virtual String GetString(byte[] bytes, int index, int count)
        {
            throw new NotImplementedException();
        }

        public virtual String GetString(byte[] bytes)
        {
            throw new NotImplementedException();
        }

        public virtual Decoder GetDecoder()
        {
            throw new NotImplementedException();
        }

        public object Clone()
        {
            throw new NotImplementedException();
        }
        
        public static Encoding UTF8
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        public static Encoding UTF8UnmarkedUnsafe
        {
            get
            {
                throw new NotImplementedException();
            }
        }
    }
}